### ISSUE TITLE
carbondump.sh doesn't get the actual cabon JVM details, insted it picks shell script running terminal details 

### ISSUE THREAD

**Issue by gvasanka** [wso2/product-ei#339]
Steps:
1. Start EI server with Oracle JDK 1.8
2. Run the carbondump.sh shell file on a terminal which has Java_Home set to IBM JDK 1.8
3. Validate the carbon dump file

Issue:
carbon_server_info.txt file shows IBM JDK details

Product                 : ${product.name} v${product.version}
WSO2 Carbon Framework   : WSO2 Carbon Framework v4.4.14
Carbon Home             : /home/asankav/servers/releasetest/ei/wso2ei-6.1.0-SNAPSHOT
Operating System Info   : Linux asankav-ThinkPad-T540p 4.4.0-62-generic #83~14.04.1-Ubuntu SMP Wed Jan 18 18:10:30 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Java Home               : /usr/local/java/ibm-java-x86_64-80
Java Version            : "1.8.0"
Java VM                 :
~

**Comment by nandika** [comment]
Does this work properly when the same jdk is used

**Comment by gvasanka** [comment]
Yes, with oracle JDK it shows correct details

**Comment by milindaperera** [comment]
@gvasanka  does carbondump tool works when both server are terminal both has Java_Home set to IBM JDK 1.8 ?

**Comment by gvasanka** [comment]
@milindaperera 
Yes, Carbondump tool works with  IBM JDK 1.8, Check below

```
asankav@asankav-ThinkPad-T540p:~/servers/releasetest/ei/23/profile/ana/wso2ei-6.1.0-SNAPSHOT/bin$ sh carbondump.sh -carbonHome /home/asankav/servers/releasetest/ei/23/profile/ana/wso2ei-6.1.0-SNAPSHOT -pid 9910

carbondump.sh##Generating the java memory dump...
carbondump.sh: 176: carbondump.sh: jmap: not found
carbondump.sh: 177: carbondump.sh: jmap: not found
carbondump.sh: 178: carbondump.sh: jmap: not found
carbondump.sh: 179: carbondump.sh: jmap: not found
carbondump.sh: 180: carbondump.sh: jmap: not found
carbondump.sh: 181: carbondump.sh: jmap: not found

carbondump.sh##Generating the thread dump...
carbondump.sh: 184: carbondump.sh: jstack: not found

carbondump.sh##Capturing OS information...
(Not all processes could be identified, non-owned process info
 will not be shown, you would have to be root to see it all.)

carbondump.sh##Capturing the list of running task in the system...

carbondump.sh##Capturing OS Environment Variables...

carbondump.sh##Generating the checksums of all the files in the CARBON_HOME directory...
md5sum: /home/asankav/servers/releasetest/ei/23/profile/ana/wso2ei-6.1.0-SNAPSHOT/repository/resources/reports/Activity: No such file or directory
md5sum: Search: No such file or directory
md5sum: Report.jrxml: No such file or directory

carbondump.sh##Copying log files...

carbondump.sh##Copying conf files...

carbondump.sh##Copying database...

carbondump.sh##Getting a directory listing...

carbondump.sh##Compressing the carbondump...
  adding: carbondump-2017-03-27_10-41-41/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/memoryinfo/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/memoryinfo/shared_object_mappings.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/memoryinfo/objects_awaiting_finalization.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/memoryinfo/java_permgen_statistics.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/memoryinfo/java_heap_histogram.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/memoryinfo/java_heap_summary.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/thread_dump.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/database/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/database/WSO2CARBON_DB.lock.db (deflated 1%)
  adding: carbondump-2017-03-27_10-41-41/repository/database/WSO2CARBON_DB.h2.db (deflated 92%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2-ei-trace.log (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/http_access_management_console_2017-03-27.log (deflated 94%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/audit.log (deflated 17%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2carbon.log (deflated 84%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2carbon.log.2017-03-23 (deflated 88%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2carbon.log.2017-03-24 (deflated 88%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2carbon-trace-messages.log (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2-ei-service.log (deflated 85%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/patches.log (deflated 85%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/audit.log.2017-03-23 (deflated 48%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/http_access_management_console_2017-03-23.log (deflated 95%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/wso2-ei-errors.log (deflated 76%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/http_access_management_console_2017-03-24.log (deflated 74%)
  adding: carbondump-2017-03-27_10-41-41/repository/logs/tm.out (deflated 95%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/multitenancy/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/multitenancy/usage-throttling-agent-config.xml (deflated 61%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/multitenancy/stratos.xml (deflated 61%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/multitenancy/cloud-services-desc.xml (deflated 83%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/multitenancy/multitenancy-packages.xml (deflated 77%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/logging-config.xml (deflated 66%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/bundle-config/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/mime.types (deflated 70%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/osgi-debug.options (deflated 71%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/launch.ini (deflated 73%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/tasks-config.xml (deflated 59%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/jmx.xml (deflated 50%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/mime.mappings (deflated 48%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/logging-bridge.properties (deflated 68%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/config-validation.xml (deflated 67%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/etc/carboncontext-osgi-services.properties (deflated 42%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/synapse-handlers.xml (deflated 42%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/data-bridge/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/data-bridge/data-agent-config.xml (deflated 74%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/registry.xml (deflated 74%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/datasources.properties (deflated 69%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tenant-mgt.xml (deflated 59%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/output-event-adapters.xml (deflated 80%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/nhttp.properties (deflated 48%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/datasources/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/datasources/master-datasources.xml (deflated 69%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/datasources/custom-datasources.xml (deflated 74%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/log4j.properties (deflated 71%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/event-broker.xml (deflated 73%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/email/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/email/email-admin-config.xml (deflated 46%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/Owasp.CsrfGuard.Carbon.properties (deflated 70%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/xss-patterns.properties (deflated 54%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/authenticators.xml (deflated 66%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/cipher-tool.properties (deflated 53%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/secret-conf.properties (deflated 69%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/cipher-text.properties (deflated 46%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/security/integrator-whitelist.properties (deflated 41%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/sec.policy (deflated 76%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/core.properties (deflated 6%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/scripts.conf (deflated 43%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/elevate.xml (deflated 47%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/stopwords.txt (deflated 43%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/lang/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/lang/stopwords_en.txt (deflated 45%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/synonyms.txt (deflated 46%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/schema.xml (deflated 74%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/currency.xml (deflated 63%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/spellings.txt (deflated 42%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/protwords.txt (deflated 46%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/registry-indexing/conf/solrconfig.xml (deflated 73%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/solr/solr.xml (deflated 51%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/synapse.properties (deflated 58%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/identity/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/identity/identity.xml (deflated 80%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/user-mgt.xml (deflated 80%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/context.xml (deflated 51%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/web.xml (deflated 85%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/carbon/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/carbon/WEB-INF/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/carbon/WEB-INF/web.xml (deflated 78%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/carbon/META-INF/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/carbon/META-INF/context.xml (deflated 47%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/catalina-server.xml (deflated 67%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/webapp-classloading-environments.xml (deflated 65%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/webapp-classloading.xml (deflated 47%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/tomcat/tomcat-users.xml (deflated 57%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/hl7.properties (deflated 39%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/access-log.properties (deflated 49%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/axis2_client.xml (deflated 79%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/axis2_pt.xml (deflated 83%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/axis2_blocking_client.xml (deflated 80%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/tenant-axis2.xml (deflated 80%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/axis2.xml (deflated 83%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/axis2/axis2_nhttp.xml (deflated 83%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/carbon.xml (deflated 70%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/passthru-http.properties (deflated 47%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/jndi.properties (deflated 46%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/rule-engine-config.xml (deflated 61%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/sample-server-wrapper.conf (deflated 68%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/README (deflated 61%)
  adding: carbondump-2017-03-27_10-41-41/repository/conf/event-processor.xml (deflated 76%)
  adding: carbondump-2017-03-27_10-41-41/directory_listing.txt (deflated 96%)
  adding: carbondump-2017-03-27_10-41-41/carbon_server_info.txt (deflated 32%)
  adding: carbondump-2017-03-27_10-41-41/checksum_values.txt (deflated 87%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/ (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_module_info.txt (deflated 85%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_open_ports.txt (deflated 88%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_installed_software_redhat_fedora.txt (stored 0%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_cpuinfo.txt (deflated 90%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_installed_software_unix_linux.txt (deflated 84%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_module_list.txt (deflated 73%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_running_tasks.txt (deflated 78%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_meminfo.txt (deflated 61%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_network_cards.txt (deflated 78%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_system_up_time.txt (deflated 48%)
  adding: carbondump-2017-03-27_10-41-41/osinfo/os_env_variables.txt (deflated 58%)

carbondump: /home/asankav/servers/releasetest/ei/23/profile/ana/wso2ei-6.1.0-SNAPSHOT/carbondump-2017-03-27_10-41-41.zip
```

**carbon_server_info.txt**

Product                 : ${product.name} v${product.version}
WSO2 Carbon Framework   : WSO2 Carbon Framework v4.4.14
Carbon Home             : /home/asankav/servers/releasetest/ei/23/profile/ana/wso2ei-6.1.0-SNAPSHOT
Operating System Info   : Linux asankav-ThinkPad-T540p 4.4.0-62-generic #83~14.04.1-Ubuntu SMP Wed Jan 18 18:10:30 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
Java Home               : /usr/local/java/ibm-java-x86_64-80
Java Version            : "1.8.0"
Java VM                 :

**Comment by milindaperera** [comment]
The server and carbondump.sh script **must** run on same JDK version (Both should be configured for Oracle JDK or IBM JDK).

**Comment by milindaperera** [comment]
carbon_server_info.txt product information retrieval is fixed in #375
